rust/crd/src/lib.rs

```diff
@@ -274,25 +348,111 @@ impl Configuration for TrinoConfig {
                     );
                     result.insert(
                         HTTP_SERVER_KEYSTORE_PATH.to_string(),
-                        Some(format!("{}/{}", KEYSTORE_DIR_NAME, "keystore.p12")),
+                        Some(format!("{}/{}", STACKABLE_CLIENT_TLS_DIR, "keystore.p12")),
                     );
-
                     result.insert(
                         HTTP_SERVER_HTTPS_KEYSTORE_KEY.to_string(),
-                        Some("secret".to_string()),
+                        Some(STACKABLE_TLS_STORE_PASSWORD.to_string()),
+                    );
+                    result.insert(
+                        HTTP_SERVER_TRUSTSTORE_PATH.to_string(),
+                        Some(format!("{}/{}", STACKABLE_CLIENT_TLS_DIR, "truststore.p12")),
+                    );
+                    result.insert(
+                        HTTP_SERVER_HTTPS_TRUSTSTORE_KEY.to_string(),
+                        Some(STACKABLE_TLS_STORE_PASSWORD.to_string()),
                     );
 
-                    // password ui login
-                    if role_name == TrinoRole::Coordinator.to_string() {
+                    if authentication_enabled {
+                        // Setting the internal secret is mandatory for authentication
                         result.insert(
-                            HTTP_SERVER_AUTHENTICATION_TYPE.to_string(),
-                            Some(HTTP_SERVER_AUTHENTICATION_TYPE_PASSWORD.to_string()),
+                            INTERNAL_COMMUNICATION_SHARED_SECRET.to_string(),
+                            Some(format!("${{ENV:{secret}}}", secret = ENV_INTERNAL_SECRET)),
                         );
+                        // For Authentication we have to differentiate several options here:
+                        // - Authentication PASSWORD: FILE | LDAP (works only with HTTPS enabled)
+                        // - requires both internal secret and client TLS to be configured
+                        if role_name == TrinoRole::Coordinator.to_string() {
+                            // password ui login
+                            result.insert(
+                                HTTP_SERVER_AUTHENTICATION_TYPE.to_string(),
+                                Some(HTTP_SERVER_AUTHENTICATION_TYPE_PASSWORD.to_string()),
+                            );
+                        }
                     }
                 }
+
+                // If internal TLS is enabled
+                if resource.get_internal_tls().is_some() {
+                    if !authentication_enabled && !client_tls_enabled {
```

Comment:
Don't we need to configure internal TLS separately from client TLS? Not sure I understand this condition...

Reply:
Its a Trino requirement https://trino.io/docs/current/security/authentication-types.html
If authentication is enabled you need secure client-server and server-server communication.
But yeah not happy with that if nesting either :D

Reply:
internal tls is not mandatory when using authentication. Only the client tls and base64 shared secret is mandatory then ;)

Reply:
yeah thats how its implemented

Reply:
But.. why? Why do we use a separate HTTPS truststore path depending on whether client TLS is enabled? Can't all of this be generic over something like `if client_tls_enabled || internal_tls_enabled`? Why do we need to do this generic TLS setup twice with slight differences?

Reply:
We offer in the crd different CAs for client and internal communication. Thats why they use different truststores.

For that if else nesting (not happy about it either):

We basically have 3 conditions that take part:
- internal-tls
- client-tls
- authentication

For Trino, all communication (client, internal) is via the same HTTP(s) port.

Now if internal communication is enabled (but e.g. no auth and client tls) we have to create the truststores etc and activate the HTTPS port for internal, but not for external access....

If only authentication is enabled, this requires client-tls encryption and a shared secret for the internal communication (if this one is not secured via tls), the HTTPS port as well as HTTP port (for internal comms)

If only client tls is enabled, this requires HTTPS port for external communication and HTTP for internal etc.

Im not sure what you mean with generic TLS setup?

Reply:
You enable TLS in almost exactly the same way twice. Instead of:

```rust
if client_tls {
  enable_tls();
  tls_truststore(server_truststore);
  tls_keystore(server_keystore);
}

if internal_tls {
  if !client_tls {
    enable_tls();
    allow_plain_text();
    tls_truststore(internal_truststore);
    tls_keystore(internal_keystore);
  }
  enable_internal_tls();
  internal_truststore(internal_truststore);
  internal_keystore(internal_keystore);
}
```

Wouldn't it be clearer to do something like:

```rust
if client_tls || internal_tls {
  enable_tls();
  let truststore, keystore;
  if client_tls {
    (truststore, keystore) = (server_truststore, server_keystore);
  } else {
    allow_plain_text();
    (truststore, keystore) = (internal_truststore, internal_keystore);
  }
  tls_truststore(server_truststore);
  tls_keystore(server_keystore);
}

if internal_tls {
  enable_internal_tls();
  internal_truststore(internal_truststore);
  internal_keystore(internal_keystore);
}
```

That way, we keep the TLS server settings grouped up, make it clear from the structure that the using the internal truststore is a fallback, and enforce that these options are set exactly once.

Alternately, if `server_truststore` doesn't actually need to contain the internal communication keys, we could just set up an empty trust-/keystore if server TLS is disabled?

Reply:
done as proposed